test/start-stop.js

```diff
@@ -0,0 +1,232 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ["error", 8] */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const async = require('async')
+const fs = require('fs')
+const path = require('path')
+const os = require('os')
+const isrunning = require('is-running')
+
+const isWindows = os.platform() === 'win32'
+const findIpfsExecutable = require('../src/utils').findIpfsExecutable
+const tempDir = require('../src/utils').tempDir
+
+const DaemonFactory = require('../src')
+
+module.exports = (type) => {
+  return () => {
+    const df = DaemonFactory.create({ type })
+    describe('starting and stopping', () => {
+      if (isWindows) {
+        return
+      }
+
+      let ipfsd
+
+      describe(`create and init a node (ipfsd)`, function () {
+        this.timeout(50 * 1000)
+        before((done) => {
+          df.spawn({ init: true, start: false, disposable: true }, (err, daemon) => {
+            expect(err).to.not.exist()
+            expect(daemon).to.exist()
+
+            ipfsd = daemon
+            done()
+          })
+        })
+
+        it('should return a node', () => {
+          expect(ipfsd).to.exist()
+        })
+
+        it('daemon should not be running', (done) => {
+          ipfsd.pid((pid) => {
+            expect(pid).to.not.exist()
+            done()
+          })
+        })
+      })
+
+      let pid
```

Comment:
Same as with `id` before. Shouldn't the `describe()`s be independent?

Reply:
see previous comment.